src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
```

Comment:
😆 let's just shoehorn that entire thing into the exception message! :trollface: 

Hmm... this is a real beard-scratcher.

OK, off the top of my head, how about a different approach:

> The direct routing topology might not deliver message types with these interfaces in their inheritance hierarchy: {list of interfaces}. Message types used for publishing should not implement these interfaces.